presto-main/src/main/java/com/facebook/presto/sql/gen/RowConstructorCodeGenerator.java

```diff
@@ -0,0 +1,78 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.gen;
+
+import com.facebook.presto.bytecode.BytecodeBlock;
+import com.facebook.presto.bytecode.BytecodeNode;
+import com.facebook.presto.bytecode.Scope;
+import com.facebook.presto.bytecode.Variable;
+import com.facebook.presto.bytecode.control.IfStatement;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.block.InterleavedBlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.relational.RowExpression;
+
+import java.util.List;
+
+import static com.facebook.presto.bytecode.expression.BytecodeExpressions.constantFalse;
+import static com.facebook.presto.bytecode.expression.BytecodeExpressions.constantInt;
+import static com.facebook.presto.bytecode.expression.BytecodeExpressions.newInstance;
+import static com.facebook.presto.sql.gen.BytecodeUtils.loadConstant;
+import static com.facebook.presto.sql.gen.SqlTypeBytecodeExpression.constantType;
+
+public class RowConstructorCodeGenerator
+        implements BytecodeGenerator
+{
+    @Override
+    public BytecodeNode generateExpression(Signature signature, BytecodeGeneratorContext context, Type rowType, List<RowExpression> arguments)
+    {
+        BytecodeBlock block = new BytecodeBlock().setDescription("Constructor for " + rowType.toString());
+        CallSiteBinder binder = context.getCallSiteBinder();
+        Scope scope = context.getScope();
+        List<Type> typeParameters = rowType.getTypeParameters();
+        int cardinality = arguments.size();
+
+        block.comment("BlockBuilder blockBuilder = new InterleavedBlockBuilder(types, new BlockBuilderStatus(), n);");
+        Variable blockBuilderVar = scope.createTempVariable(BlockBuilder.class);
+        Binding typesBinding = binder.bind(typeParameters, List.class);
+        block.append(blockBuilderVar.set(
+                newInstance(InterleavedBlockBuilder.class, loadConstant(typesBinding), newInstance(BlockBuilderStatus.class), constantInt(cardinality))));
+
+        for (int i = 0; i < cardinality; ++i) {
+            Type elementType = typeParameters.get(i);
+            Class javaType = elementType.getJavaType();
+            if (javaType == void.class) {
+                block.comment(i + "-th element type of row is undefined");
+                block.append(blockBuilderVar.invoke("appendNull", BlockBuilder.class).pop());
+            }
+            else {
+                Variable elementVar = scope.createTempVariable(javaType);
```

Comment:
We try to avoid abbreviations in the code base, so I would call this `fieldValue` instead of `elementVar`